src/cli/commands/check.js

```diff
@@ -39,108 +40,90 @@ export async function verifyTreeCheck(
   const registryName = 'yarn';
   const registry = config.registries[registryName];
   const rootManifest = await config.readManifest(registry.cwd, registryName);
+  const rootModulesFolder = config.modulesFolder || path.join(registry.cwd, registry.folder);
```

Comment:
I think we should just make `condfig.modulesFolder` to hold the ultimate truth about this.